pkg/mattermost/database_external.go

```diff
@@ -0,0 +1,115 @@
+package mattermost
+
+import (
+	"fmt"
+
+	mattermostv1beta1 "github.com/mattermost/mattermost-operator/apis/mattermost/v1beta1"
+	"github.com/mattermost/mattermost-operator/pkg/database"
+	corev1 "k8s.io/api/core/v1"
+)
+
+type ExternalDBConfig struct {
+	secretName         string
+	dbType             string
+	hasReaderEndpoints bool
+	hasDBCheckURL      bool
+}
+
+func NewExternalDBInfo(mattermost *mattermostv1beta1.Mattermost, secret corev1.Secret) (*ExternalDBConfig, error) {
```

Comment:
```suggestion
func New ExternalDBConfig(mattermost *mattermostv1beta1.Mattermost, secret corev1.Secret) (*ExternalDBConfig, error) {
```

Should this function name match the struct it creates?